Enabling and extending logging capabilities of csm restd

lasch opened the pull request:
This enables the restd logging as intended by the restd config file.

The logging was running unconfigured in existing code because the setup code was commented out and some parts were using very outdated ways to find and configure the logging. This has been fixed and therefore logging is enabled now.

The PR also adds several new log lines to help debugging problems.

lasch commented:
The existing daemon is able to reconnect to csm in case the csmd got restarted or comes back up after a shutdown. You'll see an error message indicating that the lib is not initialized and it's trying to do that again now.

The only case that's not covered yet: starting csmrestd without csmd running. This will fail and exit after initial timeout.

pdlun92 commented:
On event create (I used the shipped `/opt/ibm/csm/share/rest_scripts/spectrum_scale/create_node_leave_event.sh` script to generate this) with debug level logging I was able to observe the following in `csmrestd.log`
```
2018-11-01 16:09:34.263696   csmrestd::info     | RestApiServer: Accepted new connection.
2018-11-01 16:09:34.263935   csmrestd::debug    | csmRasEventCreate: Start processing.
2018-11-01 16:09:34.264046   csmrestd::debug    | csmRasEventCreate: msgID=spectrumscale.node.nodeLeavetime=; location=compute001
2018-11-01 16:09:34.265688   csmrestd::debug    | csmRasEventCreate: Complete.
```
However, I did not see any output to log at all on restart of csm master or aggregator daemons.  

If we think that is sufficient, let me know and I will approve. If we want to see some disconnect/reconnect activity in the log though, let me know if I am doing anything wrong

PS - passes regression as well

lasch commented:
Reworked the logging as discussed and adjusted to make it look similar/equal to what @besawn uses in csmd:

```
[RESTD]2018-11-05 14:55:34.970448   csmrestd::info     | RestApiServer: Accepted new connection.
[RESTD]2018-11-05 14:55:34.971449   csmrestd::debug    | csmRasEventCreate: Start processing.
[RESTD]2018-11-05 14:55:34.971538   csmrestd::info     | NEW RAS EVENT         ctx: ts:2018-11-10 16:53:36.05321 loc:compute01 msg:csm.status.down
[RESTD]2018-11-05 14:55:34.976002   csmrestd::info     | RAS EVENT COMPLETE    ctx: ts:2018-11-10 16:53:36.05321 loc:compute01 msg:csm.status.down
[RESTD]2018-11-05 14:55:54.214156   csmrestd::info     | RestApiServer: Accepted new connection.
[RESTD]2018-11-05 14:55:54.215145   csmrestd::debug    | csmRasEventCreate: Start processing.
[RESTD]2018-11-05 14:55:54.215264   csmrestd::info     | NEW RAS EVENT         ctx: ts:2018-11-10 16:53:36.05321 loc:compute01 msg:csm.status.down
[RESTD]2018-11-05 14:56:39.287166   csmrestd::error    | RAS EVENT ERROR       ctx: ts:2018-11-10 16:53:36.05321 loc:compute01 msg:csm.status.down errstr:Send or Recv Error
```
First event shows successful RAS event. Second shows error if csmd is down/not reachable.
